ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/Eureka2EnabledNIWSServerList.java

```diff
@@ -0,0 +1,211 @@
+package com.netflix.niws.loadbalancer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.netflix.appinfo.AmazonInfo;
+import com.netflix.appinfo.AmazonInfo.MetaDataKey;
+import com.netflix.appinfo.DataCenterInfo.Name;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.InstanceInfo.InstanceStatus;
+import com.netflix.client.config.CommonClientConfigKey;
+import com.netflix.client.config.DefaultClientConfigImpl;
+import com.netflix.client.config.IClientConfig;
+import com.netflix.config.ConfigurationManager;
+import com.netflix.eureka2.Server;
+import com.netflix.eureka2.client.EurekaInterestClient;
+import com.netflix.eureka2.client.Eurekas;
+import com.netflix.eureka2.client.resolver.ServerResolver;
+import com.netflix.eureka2.client.resolver.ServerResolvers;
+import com.netflix.eureka2.eureka1.utils.ServerListReader;
+import com.netflix.eureka2.interests.Interests;
+import com.netflix.eureka2.transport.EurekaTransports;
+import com.netflix.loadbalancer.AbstractServerList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Tomasz Bak
+ */
+public class Eureka2EnabledNIWSServerList extends AbstractServerList<DiscoveryEnabledServer> {
+
+    private static final Logger logger = LoggerFactory.getLogger(Eureka2EnabledNIWSServerList.class);
+
+    public static final String EUREKA2_WRITE_CLUSTER_HOST = "eureka2.writeCluster.host";
+    public static final CommonClientConfigKey<String> EUREKA2_WRITE_CLUSTER_HOST_KEY = new CommonClientConfigKey<String>(EUREKA2_WRITE_CLUSTER_HOST) {
+    };
+
+    public static final String EUREKA2_WRITE_CLUSTER_INTEREST_PORT = "eureka2.writeCluster.interestPort";
+    public static final CommonClientConfigKey<Integer> EUREKA2_WRITE_CLUSTER_INTEREST_PORT_KEY = new CommonClientConfigKey<Integer>(EUREKA2_WRITE_CLUSTER_INTEREST_PORT) {
+    };
+
+    public static final String EUREKA2_READ_CLUSTER_VIP = "eureka2.readCluster.vip";
+    public static final CommonClientConfigKey<String> EUREKA2_READ_CLUSTER_VIP_KEY = new CommonClientConfigKey<String>(EUREKA2_READ_CLUSTER_VIP) {
+    };
+
+    private IClientConfig clientConfig;
+
+    private String clientName;
+    private String vipAddresses;
+    private boolean isSecure;
+
+    private boolean prioritizeVipAddressBasedServers = true;
+
+    private String datacenter;
+
+    private int overridePort = DefaultClientConfigImpl.DEFAULT_PORT;
+    private boolean shouldUseOverridePort;
+    private boolean shouldUseIpAddr;
+
+    private AtomicReference<ServerListReader> serverListReaderRef = new AtomicReference<ServerListReader>();
+
+    @Override
+    public void initWithNiwsConfig(IClientConfig clientConfig) {
```

Comment:
This method is the same code as what's in DiscoveryEnabledNIWSServerList. Can we consolidate commonalities in both into a base abstract class?